test/manual/api/improved_api_tests.sh

```
@@ -0,0 +1,683 @@
+      echo "✅ Batch deleted $deleted_count/${#system_array[@]} systems"
+      # Clear the systems file
+      > "$SYSTEMS_FILE"
+      CREATED_SYSTEM_IDS=""
+    else
+      echo "❌ Batch delete failed. Status: $status"
+      
+      # Fall back to individual deletes
+      echo "Falling back to individual system deletions..."
+      local delete_count=0
+      
+      for system_id in ${system_array[@]}; do
```

Shellcheck (0.10.0)

[error] Double quote array expansions to avoid re-splitting elements.

(SC2068)